lib/validators/didErrors.js

```diff
@@ -0,0 +1,23 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
+//FIXME these errors should probably be in did-method-key
+
+export class InvalidDid extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'InvalidDidError';
+    this.code = 'invalidDid';
```

Comment:
```suggestion
    this.name = 'InvalidDidError';
    this.code = 'invalidDid';
```

Why do we have an error class whose name and code differ only in the first character? Where is `this.name` used, when an exception is thrown... in which case, would it be easier to read if this were more human readable? This same question applies to everything below.

Reply:
Ok so `name` is a property on any `Error` in javascript. 

```js
TypeError().name
"TypeError"
```

Those names follow the convention above, they as usually something like `SpecificError`.

On the other hand the error names from the spec following this pattern `specificError`.

So I decided to use not buck trends and use typical javascript Error names and then use code for the errors from the spec. I also might add that uses code makes our errors less likely to be mutated in ways that could create problems down the road.

Reply:
Seems like maybe it should be a `SyntaxError` or something instead, not a new error name?

Reply:
We could handle this like this:

```js
class DidKeyError extends Error {
  constructor(message, code) {
    super(message);
    this.name = 'DidKeyError';
    ths.code = code;
  }
}
```

This would be similar to the jsonld error pattern: https://github.com/digitalbazaar/jsonld.js/blob/main/lib/JsonLdError.js

Reply:
@msporny So we're now using `DidResolverError` or `DataError` where needed. Can this be considered resolved?